core-compose/src/main/java/com/squareup/workflow/ui/compose/ComposeViewFactory.kt

```diff
@@ -78,26 +81,50 @@ internal class ComposeViewFactory<RenderingT : Any>(
     // Composable function, so we need to use ViewGroup.setContent.
     val composeContainer = FrameLayout(contextForNewView)
 
-    // This model will be used to feed state updates into the composition.
+    // Create a single MutableState to feed state updates into the composition.
+    // We could also have two separate MutableStates, but using a Pair both makes it clear and
+    // enforces that both values are always updated together.
     val renderState = mutableStateOf<Pair<RenderingT, ViewEnvironment>?>(
+        // This will be updated immediately by bindShowRendering below.
         value = null,
         areEquivalent = StructurallyEqual
     )
 
-    // Entry point to the composition.
+    // If this is the first ever compose call in this process, we need to manually open a frame
+    // before updating the MutableState.
+    ensureComposeFrame {
```

Comment:
I think this should actually just be a single call to `FrameManager.ensureStarted()`. Don't need the wrapper function or anything.

Reply:
Yep that works!